src/core/include/openvino/core/type/element_type.hpp

```diff
@@ -126,6 +125,9 @@ class OPENVINO_API Type {
     // Return element type in string representation
     std::string to_string() const;
 
+    OPENVINO_DEPRECATED("Don't use this ctor, it's ambiguous and will be removed in 2025.0")
```

Comment:
Do we have standard template message for functionality deprecation? It should be sort of helper with deprecation date and release date arguments.

Reply:
I don't know if there is template depracated message defined. But can be added as comment of `OPENVINO_DEPRECATED` macro where example messages can be defined.

`message = What is deprecated.[Use alternative instead or reason]. Release number when it will be removed.`

e.g.
`OPENVINO_DEPRECATED("The class MyClass is deprecated. Will be removed in 2025.0 release");`

`OPENVINO_DEPRECATED("The function my_old_func() is deprecated. Use new_func() instead. Will be removed in YYYY.N release");`


```suggestion
    OPENVINO_DEPRECATED("This constructor is deprecated. Will be removed in 2025.0");
```